tests/integration/main.rs

```diff
@@ -0,0 +1,48 @@
+mod artichoke;
+
+use std::env;
+use std::path::{Path, PathBuf};
+use std::process::Command;
+
+#[allow(dead_code)]
+#[derive(Debug)]
+struct CommandOutput<'a> {
+    call_args: Vec<&'a str>,
+    status: i32,
+    stdout: String,
+    stderr: String,
+}
+
+fn manifest_path() -> String {
+    env::var_os("CARGO_MANIFEST_DIR").unwrap().into_string().unwrap()
+}
+
+fn binary_path<'a>(name: &'a str) -> Result<PathBuf, String> {
+    let path = Path::new(&manifest_path()).join("target").join("debug").join(name);
+    match path.exists() {
+        true => Ok(path),
+        false => Err(format!("Can't find binary {} in ./target/debug/", name)),
+    }
+}
+
+fn run<'a>(binary_name: &'a str, call_args: Vec<&'a str>) -> Result<CommandOutput<'a>, String> {
+    let binary = binary_path(binary_name)?;
+
+    let output = Command::new(&binary)
+        .args(call_args.clone())
+        .output()
+        .expect(format!("Failed to run ruby app {}", binary_name).as_str());
+
+    let status = match output.status.code() {
+        Some(code) => code,
+        None => -1,
+    };
```

Comment:
Fancy rust is fancy.

Just a thought on this piece actually, if there is no status code, I kind of see this as a Fatal Error. From my understanding, all applications should return an integer status code. Setting it to -1 is just because I've always seen positive codes (although I believe there's nothing preventing a negative), and it's just a slight testing signal "something went wrong". Thoughts?